lib/webauthn/authenticator_response.rb

```diff
@@ -105,8 +125,16 @@ def valid_user_verified?
       authenticator_data.user_verified?
     end
 
+    # @return [String, nil]
+    # @param [String, Array, nil] expected_origin
+    # Extract RP ID from origin in case rp_id is not provided explicitly
     def rp_id_from_origin(expected_origin)
-      URI.parse(expected_origin).host
+      case expected_origin
+      when Array
+        URI.parse(expected_origin.first).host if expected_origin.size == 1
+      when String
+        URI.parse(expected_origin).host
+      end
```

Comment:
Related to my comment below, if we transform the origin setter to set allowed_origins, we should just use allowed_origins in across this file and always expect to be an array. In such case, I would only handle this branch of logic:

`URI.parse(expected_origin.first).host if expected_origin.size == 1`